### DISCUSSION TITLE
Display abnormally in fragment shader like aliasing

### DISCUSSION THREAD

**Discussion by fanbinnet** [google/filament#6247] [Q&A]
I draw a road use a mesh,then deform it in vertex shader, then draw the line used by uv in fragment shader.
The line close to the camera show is fine， but the line far from the camera is not displayed OK.

Can someone help me point out exactly what the reason？
Thanks very much！
![notok](https://user-images.githubusercontent.com/3060688/198973570-725b583f-a452-42c3-8d6c-199e2440aae8.JPG)
The Road mesh
![road_mesn](https://user-images.githubusercontent.com/3060688/198973754-485cc791-cb08-4c41-a99f-f46072e8aa96.JPG)

**Comment by romainguy** [comment]
That's just aliasing. MSAA won't help you here, but FXAA might. But in your case I'd recommend doing your own anti-aliasing in the fragment shader. Given that you draw the lines based on UVs, you'll likely want to do something based on derivatives (with `fwidth()`) to smooth out the results.

**Reply by fanbinnet** [reply]
FXAA is enable，but don't have any improvement.

**Reply by romainguy** [reply]
Then you'll have to do what I described above.
